service/controller/resource/tccp/create.go

```diff
@@ -237,52 +240,714 @@ func (r *Resource) getCloudFormationTags(cr v1alpha1.Cluster) []*cloudformation.
 	return awstags.NewCloudFormation(tags)
 }
 
-func (r *Resource) newTemplateParams(ctx context.Context, cr v1alpha1.Cluster, t time.Time) (adapter.Adapter, error) {
+func newIAMPoliciesParams(ctx context.Context, cr v1alpha1.Cluster) (*template.ParamsMainIAMPolicies, error) {
 	cc, err := controllercontext.FromContext(ctx)
 	if err != nil {
-		return adapter.Adapter{}, microerror.Mask(err)
+		return nil, microerror.Mask(err)
 	}
 
-	// XXX: DEBUG
-	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("tccp.newTemplateParams(): cc.Spec.TenantCluster.TCCP.AvailabilityZones: %#v", cc.Spec.TenantCluster.TCCP.AvailabilityZones))
+	var iamPolicies *template.ParamsMainIAMPolicies
+	{
+		iamPolicies = &template.ParamsMainIAMPolicies{
+			ClusterID:         key.ClusterID(&cr),
+			EC2ServiceDomain:  key.EC2ServiceDomain(cc.Status.TenantCluster.AWS.Region),
+			MasterPolicyName:  key.PolicyNameMaster(cr),
+			MasterProfileName: key.ProfileNameMaster(cr),
+			MasterRoleName:    key.RoleNameMaster(cr),
+			RegionARN:         key.RegionARN(cc.Status.TenantCluster.AWS.Region),
+			KMSKeyARN:         cc.Status.TenantCluster.Encryption.Key,
+			S3Bucket:          key.BucketName(&cr, cc.Status.TenantCluster.AWS.AccountID),
+		}
+	}
+
+	return iamPolicies, nil
+}
```

Comment:
That is super nitpicky but we should have a bit of space between the functions. I do not mind either way but what I would appreciate a lot is consistency. Let's do either one or the other. The de facto standard is a new line between functions. When you want to change that we should agree on it and comply to it IMO. 

```suggestion
}

```

Reply:
This is still not sorted.